cpp/models/ode_secirts/parameters_io.h

```diff
@@ -0,0 +1,1158 @@
+ *
+ * @tparam Model The type of the model used.
+ * @tparam FP Floating point type (default: double).
+ *
+ * @param[in,out] model Vector of models, each representing a region, where the ICU population is updated.
+ * @param[in] path Path to the file containing DIVI data.
+ * @param[in] vregion Vector of region IDs for which the data is computed.
+ * @param[in] date Date for which the ICU data is computed.
+ * @param[in] scaling_factor_icu Scaling factor for reported ICU cases.
+ *
+ * @return An IOResult indicating success or failure.
+ */
+template <class Model, typename FP = double>
+IOResult<void> set_divi_data(std::vector<Model>& model, const std::string& path, const std::vector<int>& vregion,
+                             Date date, FP scaling_factor_icu)
+{
+    std::vector<FP> sum_mu_I_U(vregion.size(), 0);
+    std::vector<std::vector<FP>> mu_I_U{model.size()};
+    for (size_t region = 0; region < vregion.size(); region++) {
+        auto num_groups = model[region].parameters.get_num_groups();
+        for (auto i = AgeGroup(0); i < num_groups; i++) {
+            sum_mu_I_U[region] += model[region].parameters.template get<CriticalPerSevere<FP>>()[i] *
+                                  model[region].parameters.template get<SeverePerInfectedSymptoms<FP>>()[i];
+            mu_I_U[region].push_back(model[region].parameters.template get<CriticalPerSevere<FP>>()[i] *
+                                     model[region].parameters.template get<SeverePerInfectedSymptoms<FP>>()[i]);
+        }
+    }
+    std::vector<FP> num_icu(model.size(), 0.0);
+    BOOST_OUTCOME_TRY(read_divi_data(path, vregion, date, num_icu));
+
+    for (size_t region = 0; region < vregion.size(); region++) {
+        auto num_groups = model[region].parameters.get_num_groups();
+        for (auto i = AgeGroup(0); i < num_groups; i++) {
+            model[region].populations[{i, InfectionState::InfectedCriticalNaive}] =
+                scaling_factor_icu * num_icu[region] * mu_I_U[region][(size_t)i] / sum_mu_I_U[region];
+        }
+    }
+
+    return success();
+}
+
+/**
+* @brief reads population data from census data.
+* @param[in] path Path to the population data file.
+* @param[in] population_data Vector of population data entries.
+* @param[in] vregion Vector of keys representing the regions of interest.
+* @return An IOResult containing a vector of vectors, where each inner vector represents the population
+*         distribution across age groups for a specific region, or an error if the function fails.
+* @see mio::read_population_data
+* @{
+*/
+IOResult<std::vector<std::vector<double>>> read_population_data(const std::string& path,
+                                                                const std::vector<int>& vregion);
+IOResult<std::vector<std::vector<double>>> read_population_data(const std::vector<PopulationDataEntry>& population_data,
+                                                                const std::vector<int>& vregion);
+/**@}*/
```

Comment:
Please split this documentation, as path and population_data are not used by both functions. The rest can be reused.